WordPress/gutenberg
# Add visual indicator if a block is connected to block binding source

⚠️ This PR will need a slight refactor once https://github.com/WordPress/gutenberg/pull/58895 is merged. Once that's done, we'll have proper mechanism to add logic to "bound" blocks.

## What?
Add visual indicators if a block is connected to block bindng sources:
1. A "connection" icon in the block toolbar.
2. Focus color of the bound block is now `#9747FF`

Part of fixes in https://github.com/WordPress/gutenberg/issues/58978.

## Why?
To make the "bound" blocks visually distinct.

## How?
- Adding a new icon to `@wordpress/icons`
- Add logic to the hook which handles block binding registration.

## Testing Instructions
Register a new custom field however you prefer. You can use a snippet similar to this:

```PHP
register_meta(
	'post',
	'text_custom_field',
	array(
		'show_in_rest' => true,
		'single'       => true,
		'type'         => 'string',
		'default'	   => 'This is the content of the text custom field',
	)
);
register_meta(
	'post',
	'url_custom_field',
	array(
		'show_in_rest' => true,
		'single'       => true,
		'type'         => 'string',
		'default'	   => 'https://wpmovies.dev/wp-content/uploads/2023/04/goncharov-poster-original-1-682x1024.jpeg',
	)
);
```

In all of the following cases, verify that the purple focus outline is visible and that the connection icon appears both in the block toolbar and the list view.

**_Test paragraph_**
Add a paragraph with the content connected to a custom field:

```

<p>Content of the PAGE text custom field</p>

```

**_Test heading_**
Repeat the paragraph test but using a heading.

**_Test button_**
Add a button with the content connected to a custom field:
```

<div class="wp-block-buttons">
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Test</a></div>
</div>

```

## Screenshots or screencast 

<img width="1214" alt="Screenshot 2024-02-27 at 8 37 55 PM" src="https://github.com/WordPress/gutenberg/assets/5360536/1d3fbd44-4b5e-405b-97c3-3a545f4097e4">
